src/python/pants/backend/python/tasks2/pytest_run.py

```diff
@@ -486,11 +506,35 @@ def _get_target_from_test(self, test_info, targets):
     file_info = test_info['file']
     return relsrc_to_target.get(file_info)
 
+  def _ordered_targets_by_subset(self, targets):
+    # Partitions the given list of targets according to the selected
+    # partition. Returns the partition as a list of lists of targets in a
```

Comment:
I might be misunderstanding, but this seems to reverse-engineer the partitions a bit. Is this something that should be guaranteed by `PartitionTargets` instead?